libcst/codemod/commands/rename.py

```diff
@@ -0,0 +1,170 @@
+from libcst.codemod import CodemodContext, VisitorBasedCodemodCommand
+from libcst.codemod.visitors import AddImportsVisitor, RemoveImportsVisitor
+from libcst.helpers import get_full_name_for_node
+from libcst.metadata import QualifiedName, QualifiedNameProvider, QualifiedNameSource
+
+
+class RenameCommand(VisitorBasedCodemodCommand):
+    """
+    Rename all instances of a local or imported object.
+    """
+
+    DESCRIPTION: str = "Rename all instances of a local or imported object."
+
+    METADATA_DEPENDENCIES = (QualifiedNameProvider,)
+
+    @staticmethod
+    def add_args(parser: argparse.ArgumentParser) -> None:
+        parser.add_argument(
+            "--orig_module",
+            dest="orig_module",
+            help="Name of the module to rename. Leave blank if renaming a local variable.",
+        )
+        parser.add_argument(
+            "--orig_object",
+            dest="orig_object",
+            required=True,
+            help="Name of the object in module or variable to rename.",
+        )
+        parser.add_argument(
+            "--new_module",
+            dest="new_module",
+            help="New name of the module. Leave blank if renaming a local variable, or if same as orig_module.",
+        )
+        parser.add_argument(
+            "--new_object",
+            dest="new_object",
+            required=True,
+            help="New name of the object in module or variable.",
+        )
+
+    def __init__(
+        self,
+        context: CodemodContext,
+        orig_object: str,
+        new_object: str,
+        orig_module: Optional[str] = None,
+        new_module: Optional[str] = None,
+    ) -> None:
+        super().__init__(context)
+
+        self.orig_module: Optional[str] = orig_module
+        self.orig_object: str = orig_object
+        self.orig_name: str = (
+            f"{orig_module}.{orig_object}" if orig_module is not None else orig_object
+        )
+
+        self.new_module: Optional[
+            str
+        ] = new_module if new_module is not None else orig_module
+        self.new_object: str = new_object
+        self.new_name: str = (
+            f"{new_module}.{new_object}" if new_module is not None else new_object
+        )
+
+        self.source: QualifiedNameSource = (
+            QualifiedNameSource.IMPORT
+            if orig_module is not None
+            else QualifiedNameSource.LOCAL
+        )
+
+    def leave_Import(
+        self, original_node: cst.Import, updated_node: cst.Import
+    ) -> Union[cst.Import, cst.RemovalSentinel]:
+        for import_alias in original_node.names:
+            name = get_full_name_for_node(import_alias.name)
+            if name is not None and name == self.orig_module:
+                # Schedule this import to be potentially removed
+                RemoveImportsVisitor.remove_unused_import_by_node(
+                    self.context, original_node
+                )
+        return original_node
+
+    def leave_ImportFrom(
```

Comment:
I'm worried that this code will not work when the name you import is referenced in other places in code (for example if a module is imported); this code does not use `RemoveImportsVisitor`as `leave_Import` which checks that the import is unused before removing it.

Reply:
The problem with using RemoveImportVisitor is that in the scenario that the new `module` + `object` combination may be the same as the old one eg: `from fixit.common import CstLintRule` to replace `from package.module.common import CstLintRule`. In this scenario, the qualified names of the base class of `SomeRule(CstLintRule)` will be an array of both of those modules and RemoveImportsVisitor will not remove the old import since it's still being "used". There's really no way to avoid this with RemoveImportsVisitor.

Reply:
I think this code should be safe against removing any still-used import; it should only ever remove the single from-import of the name we are renaming.

Reply:
I looked at the implementation again. It only removes a `from ...` import if the fully-qualified name being imported exactly matches the fully-qualified name being renamed. 

This is why the test [below](https://github.com/Instagram/LibCST/pull/308/files#r442561556) pass - because we're renaming an object in a module but we're importing a module.

I don't think this code will remove an unreferenced module import, where we're renaming an object in the module.

For example,
```
from a import b

b.c = 1
```
where we're renaming "a.b.c" to "d.e.f"

Reply:
Looks like this case is now handled and there's a test for it!